core/internal/features_test.go

```diff
@@ -814,3 +815,25 @@ func TestIntegration_ExternalInitiator(t *testing.T) {
 	_, err = app.Store.JobRunsFor(jobRun.ID)
 	assert.NoError(t, err)
 }
+
+func TestIntegration_AuthToken(t *testing.T) {
+	app, cleanup := cltest.NewApplication(t)
+	defer cleanup()
+
+	eth := app.MockEthCallerSubscriber(cltest.Strict)
+	eth.Register("eth_chainId", app.Store.Config.ChainID())
+	app.Start()
+
+	// set up user
+	app.MustSeedUserAPIKey()
```

Comment:
Any suggestions on how to best implement this code for re-usability?

I assume I should implement it via as the NewHTTPClient(..) is in:
https://github.com/smartcontractkit/chainlink/blob/develop/core/internal/cltest/cltest.go#L336 
But why is an HTTPCleaner used?

Reply:
Seems like it's only used once in this PR. Perhaps we can punt on that for now and only rearchitect for reusability when it's needed.

Reply:
Ok thanks